Removed garbage_collect_data from __init__() functions

sachin-101 opened the pull request:
## Description
Removes `garbage_collect_data` from all `__init__` methods of Pointers. 
Fixes #90 

## Type of Change
Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation (non-breaking change which adds documentation)
- [x] Improvement (non-breaking change that improves the performance or reliability of existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?
Existing tests

## Checklist
- [x] I did follow the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md)
- [x] I did follow the [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have added tests for my changes

sachin-101 commented:
@AlanAboudib It turns out that `.send()` implemented in `Base Worker` calls the object's `create_pointer` method passing `garbage_collect_data` as an argument. So we can't remove it as of now.

AlanAboudib commented:
@sachin-101 I meant only removing it from _ _init_ _ methods, not from `create_pointer()`

AlanAboudib commented on syfertext/doc.py:
```diff
@@ -446,11 +445,7 @@ def create_pointer(
-            owner=owner,
-            id=ptr_id,
-            garbage_collect_data=garbage_collect_data,
+            location=location, id_at_location=id_at_location, owner=owner, id=ptr_id
```
please add `garbage_collect_data = True`. Maybe I was not clear enough, excuse me. I wanted it to be removed only from the `__init__` function definition. Although, the super class have it as True by default, but I prefer we send it explicitely to super, to make sure that if PySyft guys changed the default, this wont' affect us

sachin-101 replied:
The above is a constructor of `DocPointer` and if I add `garbage_collect_data = True` in it, then I can't remove it from ```__init__```, cause if I remove it from the constructor, I can't pass it while creating the object. :smile_cat:

AlanAboudib commented on syfertext/pipeline/pointers/subpipeline_pointer.py:
```diff
@@ -28,29 +27,24 @@ def __init__(
-            owner=owner,
-            id=id,
-            garbage_collect_data=garbage_collect_data,
+            location=location, id_at_location=id_at_location, owner=owner, id=id
```
Same as above

AlanAboudib commented on syfertext/pipeline/subpipeline.py:
```diff
@@ -176,33 +175,28 @@ def create_pointer(
-            owner=owner,
-            id=ptr_id,
-            garbage_collect_data=garbage_collect_data,
+            location=location, id_at_location=id_at_location, owner=owner, id=ptr_id
```
Same as above

AlanAboudib commented on syfertext/pointers/doc_pointer.py:
```diff
@@ -42,7 +41,6 @@ def __init__(
             id_at_location=id_at_location,
             owner=owner,
             id=id,
-            garbage_collect_data=garbage_collect_data,
```
same as above

AlanAboudib commented on syfertext/span.py:
```diff
@@ -290,11 +287,7 @@ def create_pointer(
-            owner=owner,
-            id=ptr_id,
-            garbage_collect_data=garbage_collect_data,
+            location=location, id_at_location=id_at_location, owner=owner, id=ptr_id
```
same as above

sachin-101 commented:
@AlanAboudib I reverted back and added the new suggested changes. So some of your review comments would have vanished from the files changed section. :smile:

AlanAboudib commented:
Any updates on this issue @sachin-101

sachin-101 commented:
> I agree on overriding the value in create_pointer methods
The overriding is done by passing `True` while initializing the Pointer Object. Hence the `garbage_collect_data` argument to the `Pointer object` constructor is independent of the value passed to the argument `garbage_collect_data` from PySyft.
And hence I haven't removed the argument from the constructor.
